src/stats/cdf.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright 2016 Johannes Köster.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Support for discrete probability distributions in terms of cumulative distribution
+//! functions (CDF).
+
+use std::f64;
+use std::iter;
+use std::slice;
+
+use num::traits::{cast, NumCast};
+use itertools::Itertools;
+
+use stats::logprobs::{self, LogProb};
+
+
+/// Implementation of a cumulative distribution function.
+#[derive(Debug, Clone)]
+pub struct CDF<T: PartialOrd> {
+    inner: Vec<(T, LogProb)>
```

Comment:
Can you make a struct with named fields for `T, LogProb`? There are lots of `.0` and `.1`s floating around-- I think it'd make the code clearer if they had names.